### PR TITLE
Categorize doctl commands for help output (APICLI-86)

### PR DESCRIPTION
**Pull request by danaelhe**
Change from this:

```
Available Commands:
  1-click         Display commands that pertain to 1-click applications
  account         Display commands that retrieve account details
  apps            Display commands for working with apps
  auth            Display commands for authenticating doctl with an account
  balance         Display commands for retrieving your account balance
  billing-history Display commands for retrieving your billing history
  completion      Modify your shell so doctl commands autocomplete with TAB
  compute         Display commands that manage infrastructure
  databases       Display commands that manage databases
  help            Help about any command
  invoice         Display commands for retrieving invoices for your account
  kubernetes      Displays commands to manage Kubernetes clusters and configurations
  projects        Manage projects and assign resources to them
  registry        Display commands for working with container registries
  version         Show the current version
  vpcs            Display commands that manage VPCs
```

to this:

```
Manage DigitalOcean resources:
  account         Display commands that retrieve account details
  compute         Display commands that manage infrastructure
  databases       Display commands that manage databases
  kubernetes      Displays commands to manage Kubernetes clusters and configurations
  projects        Manage projects and assign resources to them
  apps            Display commands for working with apps
  registry        Display commands for working with container registries
  vpcs            Display commands that manage VPCs
  1-click         Display commands that pertain to 1-click applications

Configure doctl:
  auth            Display commands for authenticating doctl with an account
  completion      Modify your shell so doctl commands autocomplete with TAB
  version         Show the current version

View DigitalOcean Billing:
  balance         Display commands for retrieving your account balance
  billing-history Display commands for retrieving your billing history
  invoice         Display commands for retrieving invoices for your account
```

Looking for feedback on the way the commands are categorized as well.

### CODE REVIEW

**Review by andrewsomething — COMMENTED** [review]
Very happy to see this!

One thing I'm noticing, is that the new output lacks any mention of the global flags. Not just at the top-level, but every sub-command as well.  From the current output:

```
Global Flags:
  -t, --access-token string   API V2 access token
  -u, --api-url string        Override default API endpoint
  -c, --config string         Specify a custom config file (default "/home/asb/.config/doctl/config.yaml")
      --context string        Specify a custom authentication context name
  -o, --output string         Desired output format [text|json] (default "text")
      --trace                 Show a log of network activity while performing a command
  -v, --verbose               Enable verbose output
```

We should make sure there are not other unintended changes as well.

I wonder if it would be possible to take some inspiration from how `k8s.io/kubectl/pkg/util/templates` implements this but not import it directly. It seems to make a number of decisions on how Cobra is used beyond just the groupings. It's not clear if it is really meant to be used as a standalone library or if it is internal to kubectl. I'm not sure we want to tie such a fundamental piece of our UX to kubectl. It also pulls in quite a few dependencies. So reducing those would be an added benefit of not importing it.

File: commands/doit.go [hunk 1]
```diff
@@ -130,21 +131,41 @@ func Execute() {
...
-	DoitCmd.AddCommand(computeCmd())
-	DoitCmd.AddCommand(Kubernetes())
-	DoitCmd.AddCommand(Databases())
-	DoitCmd.AddCommand(Projects())
-	DoitCmd.AddCommand(Version())
-	DoitCmd.AddCommand(Registry())
-	DoitCmd.AddCommand(VPCs())
-	DoitCmd.AddCommand(OneClicks())
+
+	commandGroups := templates.CommandGroups{
```
Comment (ChiefMateStarbuck) [inline commands/doit.go:135]: This is smart

File: commands/doit.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/fatih/color"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"k8s.io/kubectl/pkg/util/templates"
```
Comment (ChiefMateStarbuck) [inline commands/doit.go:27]: i second what @andrewsomething stated.

**Comment by danaelhe** [comment]
> Very happy to see this!
> 
> One thing I'm noticing, is that the new output lacks any mention of the global flags. Not just at the top-level, but every sub-command as well. From the current output:
> 
> ```
> Global Flags:
>   -t, --access-token string   API V2 access token
>   -u, --api-url string        Override default API endpoint
>   -c, --config string         Specify a custom config file (default "/home/asb/.config/doctl/config.yaml")
>       --context string        Specify a custom authentication context name
>   -o, --output string         Desired output format [text|json] (default "text")
>       --trace                 Show a log of network activity while performing a command
>   -v, --verbose               Enable verbose output
> ```
> 
> We should make sure there are not other unintended changes as well.
> 
> I wonder if it would be possible to take some inspiration from how `k8s.io/kubectl/pkg/util/templates` implements this but not import it directly. It seems to make a number of decisions on how Cobra is used beyond just the groupings. It's not clear if it is really meant to be used as a standalone library or if it is internal to kubectl. I'm not sure we want to tie such a fundamental piece of our UX to kubectl. It also pulls in quite a few dependencies. So reducing those would be an added benefit of not importing it.

I totally agree- great points. I will rework the solution to
1) Decouple ourselves from the kubectl templates (but take inspo)
2) bring back our flags!

thanks!

**Review by Davi3031 — COMMENTED** [review]
Manage DigitalOcean resources:
  account         Display commands that retrieve account details
  compute         Display commands that manage infrastructure
  databases       Display commands that manage databases
  kubernetes      Displays commands to manage Kubernetes clusters and configurations
  projects        Manage projects and assign resources to them
  apps            Display commands for working with apps
  registry        Display commands for working with container registries
  vpcs            Display commands that manage VPCs
  1-click         Display commands that pertain to 1-click applications

Configure doctl:
  auth            Display commands for authenticating doctl with an account
  completion      Modify your shell so doctl commands autocomplete with TAB
  version         Show the current version

View DigitalOcean Billing:
  balance         Display commands for retrieving your account balance
  billing-history Display commands for retrieving your billing history
  invoice         Display commands for retrieving invoices for your account

**Comment by Davi3031** [comment]
Manage DigitalOcean resources:
  account         Display commands that retrieve account details
  compute         Display commands that manage infrastructure
  databases       Display commands that manage databases
  kubernetes      Displays commands to manage Kubernetes clusters and configurations
  projects        Manage projects and assign resources to them
  apps            Display commands for working with apps
  registry        Display commands for working with container registries
  vpcs            Display commands that manage VPCs
  1-click         Display commands that pertain to 1-click applications

Configure doctl:
  auth            Display commands for authenticating doctl with an account
  completion      Modify your shell so doctl commands autocomplete with TAB
  version         Show the current version

View DigitalOcean Billing:
  balance         Display commands for retrieving your account balance
  billing-history Display commands for retrieving your billing history
  invoice         Display commands for retrieving invoices for your account

**Comment by danaelhe** [comment]
> Manage DigitalOcean resources: account Display commands that retrieve account details compute Display commands that manage infrastructure databases Display commands that manage databases kubernetes Displays commands to manage Kubernetes clusters and configurations projects Manage projects and assign resources to them apps Display commands for working with apps registry Display commands for working with container registries vpcs Display commands that manage VPCs 1-click Display commands that pertain to 1-click applications
> 
> Configure doctl: auth Display commands for authenticating doctl with an account completion Modify your shell so doctl commands autocomplete with TAB version Show the current version
> 
> View DigitalOcean Billing: balance Display commands for retrieving your account balance billing-history Display commands for retrieving your billing history invoice Display commands for retrieving invoices for your account

i like this organization a lot!

**Comment by ChiefMateStarbuck** [comment]
We're gonna start over!

**Pull request closed without merging**